src/System.Text.JsonLab.Serialization/System/Text/Json/Serialization/JsonConverter.Sync.cs

```diff
@@ -153,19 +138,14 @@ private static Span<byte> ToJsonInternal(object value, JsonConverterSettings set
 
             byte[] result;
 
-            using (var output = new ArrayBufferWriter(settings.DefaultBufferSize))
+            using (var output = new ArrayBufferWriter<byte>(settings.DefaultBufferSize))
             {
                 var writer = new Utf8JsonWriter(output, state);
 
                 ToJson(ref writer, settings, ref current, ref previous, ref arrayIndex);
 
-                int byteCount;
-                checked
-                {
-                    byteCount = (int)writer.BytesWritten;
-                }
-
-                Span<byte> json = output.GetSpan().Slice(0, byteCount);
+                writer.Flush(isFinalBlock: true);
+                ReadOnlyMemory<byte> json = output.WrittenMemory;
```

Comment:
Can we avoid calling ToArray on the written memory some how? For the string case, we should just transcode and return the string to avoid the extra allocation.

Span is trickier. As an aside, one thing I noticed, shouldn't we return a `ReadOnlySpan<byte>` instead?